client/src/app/_services/permission.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Permission } from '../_models/index';
+
+import 'rxjs/add/observable/throw';
+import 'rxjs/add/operator/catch';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class PermissionService {
+
+    constructor(private http: HttpClient) { }
+
+    getAll() {
+        return this.http.get<Permission[]>('/api/permissions').catch(this.handleError);
+    }
+
+    getById(id: Number) {
```

Comment:
This service does not have anything else than get all function

Reply:
Yes you are right!